src/ImmersedBoundaries/conditional_fluxes.jl

```diff
@@ -44,25 +97,25 @@ const ATD = AbstractTimeDiscretization
 
 # dx(uu), dy(vu), dz(wu)
 # ccc,    ffc,    fcf
-@inline _advective_momentum_flux_Uu(i, j, k, ibg::GFIBG, args...) = conditional_flux_ccc(i, j, k, ibg, advective_momentum_flux_Uu, args...)
-@inline _advective_momentum_flux_Vu(i, j, k, ibg::GFIBG, args...) = conditional_flux_ffc(i, j, k, ibg, advective_momentum_flux_Vu, args...)
-@inline _advective_momentum_flux_Wu(i, j, k, ibg::GFIBG, args...) = conditional_flux_fcf(i, j, k, ibg, advective_momentum_flux_Wu, args...)
+@inline _advective_momentum_flux_Uu(i, j, k, ibg::GFIBG, closure, U, args...) = conditional_flux_ccc(i, j, k, ibg, advective_momentum_flux_Uu, closure, U, args...)
+@inline _advective_momentum_flux_Vu(i, j, k, ibg::GFIBG, closure, U, args...) = conditional_flux_ffc(i, j, k, ibg, advective_momentum_flux_Vu, closure, U, args...)
+@inline _advective_momentum_flux_Wu(i, j, k, ibg::GFIBG, closure, U, args...) = conditional_flux_fcf(i, j, k, ibg, advective_momentum_flux_Wu, closure, U, args...)
```

Comment:
Depending on how I define the arguments of this function, I get either an error when calling `div_𝐯u` (with a looong stack trace and no obvious error) or this just calls the default `_advective_momentum_flux_Wu(...) = advective_momentum_flux_Wu(...)` method...

I'm not sure what I'm getting wrong here, but suggestions are welcome.

Reply:
hmmm weird, this should call 
`advective_momentum_flux_Wu(i, j, k, grid, scheme, W, u)` (so I guess your `closure` here is the `scheme`) or zero if on a `solid_interface`

Reply:
Thanks for the help, @simone-silvestri. It's much appreciated. Here are some comments:

> conditional_flux should work both for viscous_flux and advective_flux. So every specific argument to one of the two should go after the function specification, just i, j, k, grid before

The reason these arguments are specified is because we need `U` to be passed to the drag function as well. (@whitleyv can you confirm?)

And yes, you're right about the drag right now being counted twice. My plan was to deal with this after I got everything working, but your comment made me realize that separating `conditional_flux` into two will probably help with implementation right now since I can pick one whose arguments will be passed as `args...` :)

Reply:
Yes. There might be an easier way to get those fields, but if I didn't have all the args written out up to the velocity, it wouldn't recognize it as the velocity needed in the drag function down the line.